src/main/scala/com/kinja/amqp/AmqpClientFactory.scala

```diff
@@ -46,12 +46,12 @@ class AmqpClientFactory {
 		actorSystem: ActorSystem,
 		logger: Logger,
 		ec: ExecutionContext,
-		messageStores: ::[(AtLeastOnceGroup, MessageStore)],
+		messageStores: List[(AtLeastOnceGroup, MessageStore)],
```

Comment:
Since we allow creating a client without a message store to back it up anyway, we don't really need to require a non-empty list here.

Reply:
Summary of offline talk:
for consumer only clients we should use: https://github.com/gawkermedia/amqp-client-provider/blob/d3a6e4de16e1d7baf2e66e6875149c77f88367b1/src/main/scala/com/kinja/amqp/AmqpClientFactory.scala#L19
and for producers we always want to have messagestore.